tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -3901,9 +3909,10 @@ def __init__(self, input_dataset, batch_size, drop_remainder):
     self._drop_remainder = ops.convert_to_tensor(
         drop_remainder, dtype=dtypes.bool, name="drop_remainder")
 
-    constant_drop_remainder = tensor_util.constant_value(self._drop_remainder)
     # pylint: disable=protected-access
-    if constant_drop_remainder:
+    evenly_divisible = input_dataset.__evenly_divisible__(self._batch_size)
+    constant_drop_remainder = tensor_util.constant_value(self._drop_remainder)
+    if constant_drop_remainder or tensor_util.constant_value(evenly_divisible):
```

Comment:
In graph mode will `tensor_util.constant_value(evenly_divisible)` ever return something other than `None`?

Reply:
It was quite one month old and I need to recompile everything cause I've not RBE access as the TF team and I was on others PR/branches. I've reverted a single test change to see if it fail in graph mode.

Reply:
Ok it seems It has passed a special test in graph mode only so probably the original tests protfolio can pass with the none case in graph mode as all the tests passed.

As I don't have the cache ready for this PR please close this if you don't see any other solution to have exactly the same behaviour in graph mode with drop false.